## Betas Analysis

In [2]:
import numpy as np
from scipy.io import loadmat
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots
from numpy import genfromtxt

### Pz - Component 5 (P3b)

#### ROIs

In [3]:
'LOAD OUTPUTS FROM BETA SCRIPT'

pz_fastbeta = loadmat('ROIs/Pz_ROIs/FastBetaResults_Pz.mat')
pz_beta = loadmat('ROIs/Pz_ROIs/BetaResults_Pz.mat')

In [11]:
'EXTRACT BETAS'

'ROIs are in a numpy void class'
pz_data = pz_fastbeta['Data']
pz_data = pz_data[0]
pz_data = pz_data[0]

#### Capacity

In [5]:
'UPDATE CAPACITY SUBJECT LIST'

capacity = genfromtxt('span.csv', delimiter=',')
capacity = capacity[1:,:] # Take off the title

print('Span list length: '+str(len(capacity)))

'List of subject IDs'
subjects = pz_data[1]

updated_list = []
for i in range(len(capacity)):
    if (capacity[i,0] in subjects):
        updated_list.append(float(capacity[i,1]))
    
print('\nUpdated span list length: '+str(len(updated_list)))

Span list length: 170

Updated span list length: 133


### Region of Interest Scatterplots: WM capacity x Load effect

In [25]:
'Iterate through ROIs from component 5'
for i in range((len(pz_data)-2)): # Ignore first two index
    ROI = pz_data[i+2].T # Transpose it to access the columns
    low_load = ROI[2]
    high_load = ROI[5]  
    load_effect = high_load - low_load
    
    capacity = updated_list
    
    fig = px.scatter(x=capacity, y=load_effect, trendline="lowess")
    
    fig.update_layout(
        title="ROI "+str(i+1),
        xaxis_title="Capacity",
        yaxis_title="Load Effect",
    )
        
    fig.show()

### O1, Oz, O2 Average - Component 7 (N170)

In [12]:
'LOAD OUTPUTS FROM BETA SCRIPT'

oavg_fastbeta = loadmat('ROIs/oavg_ROIs/FastBetaResults_oavg.mat')
oavg_beta = loadmat('ROIs/oavg_ROIs/BetaResults_oavg.mat')

In [22]:
'EXTRACT LOAD EFFECTS'

'ROIs are in a numpy void class'
oavg_data = oavg_fastbeta['Data']
oavg_data = oavg_data[0]
oavg_data = oavg_data[0]

ROI = oavg_data[0+2].T
low_load = ROI[2]
print(len(low_load))

170


Uhhh.... why are there 170 participants in this analysis?